src/webhintUtils.ts

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as vscode from 'vscode';
+import * as path from 'path';
+
+import type { UserConfig as WebhintUserConfig } from '@hint/utils';
+
+export async function getWebhintUserConfig(path: vscode.Uri): Promise<WebhintUserConfig | undefined> {
+  if (!path || !path.toString()) {
+      return;
+  }
+
+  // start with an empty file
+  let userConfigByteArray: Uint8Array = Buffer.from('', 'utf8');
+
+  try {
+      // verify if .hintrc exists
+      await vscode.workspace.fs.stat(path);
+  } catch (error) {
+      if ((error as vscode.FileSystemError).code === vscode.FileSystemError.FileNotFound.name) {
+
+          // .hintrc does not exists so create one with the default config
+          const defaultConfig = { extends: ['development'] };
+          await vscode.workspace.fs.writeFile(path, Buffer.from(JSON.stringify(defaultConfig), 'utf8'));
```

Comment:
I wonder if we should do these as workspace edits to maximize VS Code's understanding of what we're doing. See this sample extension for an example: https://github.com/microsoft/vscode-extension-samples/blob/30c1740717acdc7e4aabc783d07bd83e21857320/code-actions-sample/src/extension.ts#L74